public/pages/AnomalyCharts/containers/AnomalyHeatmapChart.tsx

```diff
@@ -0,0 +1,519 @@
+        },
+      ];
+    };
+
+    const [originalHeatmapData, setOriginalHeatmapData] = useState(
+      getAnomaliesHeatmapData(
+        props.anomalies,
+        props.dateRange,
+        AnomalyHeatmapSortType.SEVERITY,
+        COMBINED_OPTIONS.options[0].value
+      )
+    );
+    const [heatmapData, setHeatmapData] = useState<PlotData[]>(
+      originalHeatmapData
+    );
+
+    const [sortByFeildValue, setSortByFeildValue] = useState(
+      SORT_BY_FIELD_OPTIONS[0].value
+    );
+
+    const [currentViewOptions, setCurrentViewOptions] = useState([
+      getViewEntityOptions(originalHeatmapData)[0].options[0],
+    ]);
+
+    const [entityViewOptions, setEntityViewOptions] = useState(
+      getViewEntityOptions(originalHeatmapData)
+    );
+
+    const [showAlertsFlyout, setShowAlertsFlyout] = useState<boolean>(false);
+
+    const [numEntities, setNumEntities] = useState(
+      originalHeatmapData[0].y.length
+    );
+
+    const hasAnomalyInHeatmap = (): boolean => {
+      for (let entityAnomaliesOccurence of heatmapData[0].text) {
+        //@ts-ignore
+        const sum = entityAnomaliesOccurence.reduce((a, b) => {
+          return a + b;
+        });
+        if (sum > 0) {
+          return true;
+        }
+      }
+      return false;
+    };
+
+    const handleHeatmapClick = (event: Plotly.PlotMouseEvent) => {
+      const selectedCellIndices = get(event, 'points[0].pointIndex', []);
+      const selectedEntity = get(event, 'points[0].y', '');
+      if (!isEmpty(selectedCellIndices)) {
+        let anomalyCount = get(event, 'points[0].text', 0);
+        if (
+          anomalyCount === 0 ||
+          (heatmapData.length > 1 &&
+            //@ts-ignore
+            heatmapData[1].z[selectedCellIndices[0]][selectedCellIndices[1]] !=
+              null)
+        ) {
+          // in the case of re-clicking on selected cell
+          const noCellSelectedHeatmapData = updateHeatmapPlotData(
+            heatmapData[0],
+            {
+              opacity: 1,
+            }
+          );
+          setHeatmapData([noCellSelectedHeatmapData]);
+          props.onHeatmapCellSelected(undefined);
+        } else {
+          const transparentHeatmapData = updateHeatmapPlotData(heatmapData[0], {
+            opacity: 0.3,
+          });
+
+          const selectedHeatmapData = getSelectedHeatmapCellPlotData(
+            heatmapData[0],
+            selectedCellIndices[1],
+            selectedCellIndices[0]
+          );
+          setHeatmapData([transparentHeatmapData, ...selectedHeatmapData]);
+
+          const selectedEndDate = moment(
+            //@ts-ignore
+            heatmapData[0].x[selectedCellIndices[1]],
+            HEATMAP_X_AXIS_DATE_FORMAT
+          ).valueOf();
+
+          const selectedStartDate =
+            selectedEndDate -
+            getHeatmapCellDateRangeInterval(
+              //@ts-ignore
+              heatmapData[0].x,
+              selectedCellIndices[1]
+            );
```

Comment:
seems confusing to be taking an end date and subtracting with an interval, maybe rename `Interval` in `getHeatmapCellDateRangeInterval` to something like `Size` or `Length`

Reply:
agree. changing to `Length`.